src/main/java/bio/terra/service/auth/iam/sam/SamApiService.java

```diff
@@ -24,10 +26,18 @@ public class SamApiService {
   private final OkHttpClient sharedHttpClient;
 
   @Autowired
-  public SamApiService(SamConfiguration samConfig, ConfigurationService configurationService) {
+  public SamApiService(
+      SamConfiguration samConfig,
+      ConfigurationService configurationService,
+      OpenTelemetry openTelemetry) {
     this.samConfig = samConfig;
     this.configurationService = configurationService;
-    this.sharedHttpClient = new ApiClient().getHttpClient();
+    this.sharedHttpClient =
+        new ApiClient()
+            .getHttpClient()
+            .newBuilder()
+            .addInterceptor(new OkHttpClientTracingInterceptor(openTelemetry))
```

Comment:
(I am making this comment in the middle of my review, so it may be a moot point!)

Wondering if we also need to add interceptors to the other external clients we build (ECM, etc)?

Reply:
Yes, I think that would be great! I've added this as a part of the follow-on work for otel. https://broadworkbench.atlassian.net/browse/DR-3456